salt/transport/tcp.py

```diff
@@ -935,6 +1125,157 @@ def publish_payload(self, package, topic_list=None):
+                                    socket to bind to. This socket does
+                                    not need to exist prior to calling
+                                    this method, but parent directories
+                                    should.
+                                    It may also be of type 'int', in
+                                    which case it is used as the port
+                                    for a tcp localhost connection.
+        :param IOLoop io_loop: A Tornado ioloop to handle scheduling
+        :param func payload_handler: A function to customize handling of
+                                     incoming data.
+        """
+        self.socket_path = socket_path
+        self._started = False
+        self.payload_handler = payload_handler
+
+        # Placeholders for attributes to be populated by method calls
+        self.sock = None
+        self.io_loop = io_loop or tornado.ioloop.IOLoop.current()
+        self._closing = False
+
+    def start(self):
+        """
+        Perform the work necessary to start up a Tornado IPC server
+
+        Blocks until socket is established
+        """
+        # Start up the ioloop
+        log.trace("IPCServer: binding to socket: %s", self.socket_path)
+        if isinstance(self.socket_path, int):
+            self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            self.sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            self.sock.setblocking(0)
+            self.sock.bind(("127.0.0.1", self.socket_path))
+            # Based on default used in tornado.netutil.bind_sockets()
+            self.sock.listen(128)
+        else:
+            self.sock = tornado.netutil.bind_unix_socket(self.socket_path)
+
+        tornado.netutil.add_accept_handler(
+            self.sock,
+            self.handle_connection,
+        )
+        self._started = True
+
+    async def handle_stream(self, stream):
+        """
+        Override this to handle the streams as they arrive
+
+        :param IOStream stream: An IOStream for processing
+
+        See https://tornado.readthedocs.io/en/latest/iostream.html#tornado.iostream.IOStream
+        for additional details.
+        """
+
+        async def _null(msg):
+            return
+
+        def write_callback(stream, header):
+            if header.get("mid"):
+
+                async def return_message(msg):
+                    pack = salt.transport.frame.frame_msg_ipc(
+                        msg,
+                        header={"mid": header["mid"]},
+                        raw_body=True,
+                    )
+                    await stream.write(pack)
+
+                return return_message
+            else:
+                return _null
+
+        unpacker = salt.utils.msgpack.Unpacker(raw=False)
+        while not stream.closed():
+            try:
+                wire_bytes = await stream.read_bytes(4096, partial=True)
```

Comment:
magic numbers

Reply:
We're going with what the value has been since originally written. Unless there is a compelling need to change, I'd prefer not too unless it's extremely well tested. That said, it falls outside of the scope of this PR.